sdks/java/extensions/sql/build.gradle

```diff
@@ -45,23 +48,27 @@ dependencies {
   compile "org.apache.calcite:calcite-core:$calcite_version"
   compile "org.apache.calcite:calcite-linq4j:$calcite_version"
   compile "org.apache.calcite.avatica:avatica-core:$avatica_version"
-  compile "com.alibaba:fastjson:1.2.12"
   compileOnly library.java.findbugs_jsr305
   compileOnly library.java.findbugs_annotations
   shadow project(path: ":beam-sdks-java-core", configuration: "shadow")
   shadow project(path: ":beam-sdks-java-extensions-join-library", configuration: "shadow")
   shadow library.java.slf4j_api
+  shadow library.java.slf4j_jdk14
   shadow library.java.commons_csv
+  shadow library.java.commons_lang3
+  shadow library.java.jackson_databind
   shadow library.java.joda_time
+  shadow "com.alibaba:fastjson:1.2.12"
   shadow project(path: ":beam-runners-direct-java", configuration: "shadow")
   provided project(path: ":beam-sdks-java-io-kafka", configuration: "shadow")
   provided project(path: ":beam-sdks-java-io-google-cloud-platform", configuration: "shadow")
   provided library.java.kafka_clients
-  testCompile library.java.slf4j_jdk14
   testCompile library.java.junit
   testCompile library.java.hamcrest_core
   testCompile library.java.mockito_core
   testCompile library.java.quickcheck_core
+  shadowJarTest project(path: project.path, configuration: "shadow")
```

Comment:
A question more for @lukecwik than you: shouldn't all of our tests run against the actual jar produced for users? Is the shadow configuration that jar? Or is it a different one?

Reply:
Just setting the classpath to `configurations.shadowTest` didn't work, because it didn't include the output jar. This works, but a good number of tests fail everywhere due to dependency issues. They are all probably real issues.

```
  configurations {
    validate
  }

  dependencies {
    validate project(path: project.path, configuration: "shadowTest")
  }

  test {
    classpath = configurations.validate
  }
```

cc: @boyuanzz Who might be working on the same issue.

Reply:
Is producing the `shadowJar/shadowTestJar` a dependency of the `test` task?